src/interface/state_descriptor.hpp

```diff
@@ -64,6 +65,28 @@ class StateDescriptor {
   // retrieve label
   const std::string &label() { return label_; }
 
+  bool AddSwarm(const std::string &swarm_name, Metadata &m) {
+    if (swarmMetadataMap_.count(swarm_name) > 0) {
+      throw std::invalid_argument("Swarm " + swarm_name + " already exists!");
+    }
+    swarmMetadataMap_[swarm_name] = m;
```

Comment:
So it looks like swarms and swarmvalues can have different metadata. How does this work? Must the swarmvalue metadata contain the swarm metadata?

Reply:
I'm not doing much with metadata at the moment, though I assume it will be needed more later (or even now?) Right now the swarmvalue metadata just says whether its Real-valued or integer-valued. The Swarm metadata doesn't do anything.

Reply:
Huh. Yeah, so metadata matters when you want to generically operate over some subset of particles/swarm. For example, you can use metadata to:
- Mark what particles/variables need boundary communication
- Mark what particles/variables should be evolved, and what should be sampled independently per timestep

Stuff like that. I'm just throwing ideas out off the top of my head. Maybe it's worth revisiting later after we have some experience application-wise. Just worth keeping in mind as a capability and not coding ourselves out of using it.

Reply:
Yes I think what you say makes sense. But I think this should probably be deferred for a subsequent PR when these things are added.

Reply:
Yeah that's fine.